moon-store-mac-os/moon-store-mac-os/Views/Product/ProductListView.swift

```diff
@@ -40,174 +35,181 @@ private enum ProductTableTitle: String, CaseIterable {
     case inStock
     case price
     case options
-    
+
     var title: String {
         switch self {
-            case .product: "Producto"
-            case .category: "Categoría"
-            case .inStock: "En Stock"
-            case .price: "Precio"
-            case .options, .productIcon: ""
+        case .product: "Producto"
+        case .category: "Categoría"
+        case .inStock: "En Stock"
+        case .price: "Precio"
+        case .options, .productIcon: ""
         }
     }
 }
 
 struct ProductListView: View {
-    @State private var searchValue: String = ""
-    
+    @StateObject private var viewModel: ProductListViewModel = .init()
```

Comment:
Si este ViewModel no se injecta desde otro lado, deberia de ser `@ObservedObject`

Reply:
De hecho estas en lo cierto este viewModel no es inyectado desde fuer por eso usamos StateObject. La instancia del viewModel vive solo en esta vista.

https://www.hackingwithswift.com/quick-start/swiftui/what-is-the-stateobject-property-wrapper